aten/src/ATen/native/cuda/UpSample.cuh

```diff
@@ -163,6 +163,15 @@ __device__ __forceinline__ static int nearest_neighbor_compute_source_index(
   return src_index;
 }
 
+__device__ __forceinline__ static int nearest_neighbor_compute_destination_index(
+    const float scale,
+    int src_index,
+    int output_size) {
+  const int dst_index = min<int>(
```

Comment:
Do you need template specialization here?

Reply:
Was blindly following the code there and I don't see any necessity for explicit specialization here neither. I'll update the other code as well.